Adding multi-LTS Travis-CI script.

mmzx opened the pull request:
Would you mind? As a next step I would like to use your branch to fix package dependency issues and warnings.

tolysz committed:
parent 8f71e013e79bfdebe987542a7ee6f926ec9bdee4
author Marcin Tolysz <tolysz@gmail.com> 1514741013 +0000
committer Akos Marton <makos999@protonmail.ch> 1582496404 +0100

Squashed rebase from Marcin Tolysz, tolysz/modernize.

* Builds with `process` 1.6
* add dependency for `tls-session-manager`
* bump resolver

Add per bundle certificates for static files and redirects.

update sample configuration

add reverse proxy

* Builds with `process` 1.6
* add dependency for `tls-session-manager`
* bump resolver

fix reverse proxy for non http1.1 connections

Make it compile against more resolvers:
resolver: lts-13.8
resolver: lts-12.8
resolver: lts-11.8
resolver: lts-10.8

Delete stack.yaml

Update .gitignore

Update README.md

Update README.md

clean

Update setup-keter.sh

Update PortPool.hs

Update PortPool.hs

Update PortPool.hs

Update PortPool.hs

Update App.hs

Update App.hs

Update App.hs

Update App.hs

Update App.hs

Update App.hs

Update App.hs

Update App.hs

Update App.hs

Update LabelMapSpec.hs

Update LabelMapSpec.hs

Update keter.cabal

Update keter.cabal

Commit:
Updating CI script to use multiple versions of ghc for testing.

Commit:
Strict constraint on process package to have v1.4.3.0.

Commit:
Strict constraint on process package to have v1.4.3.0.

Commit:
Experimenting with TravisCI.

Commit:
Adding upper bound on fsnotify package.

Commit:
Adding upper bound on fsnotify package.

Commit:
Adopting network-conduit api change.

Commit:
Adopting network-conduit api change.

Commit:
Use cabal constraints.

Commit:
Adding stack version of CI.

Commit:
unix-compat version bounds

Commit:
unix-compat version bounds

Commit:
Complex travis script.

Commit:
Removing osx build target.

Commit:
Update demanded lts versions.

Commit:
Cabal builds and ghc-head are not needed for now.

Commit:
Adding multi-LTS Travis-CI script.

Adding multi-LTS Travis-CI script.

tolysz commented:
Nice. Could you _just_ rebase your changes on top of the "modernize" branch? I have a feeling, from the file changed, that this one is removing my changes.
I.e. I would expect to see only changes to `.travis.yaml` or some build script.

mmzx commented:
This would be okay to start with, based on the results of CI - I would plan like adding more compilers, package sets and every further action would need to have a pass.
This way it will also be easy to see what compilers and/or package cannot be further supported.

Commit:
Merge branch 'ciScript' of github.com:mmzx/keter into ciScript

Commit:
Adding cabal build target to CI.

mmzx commented:
I will have to apply your changes one-by-one, migrating to newer version of the _process_ library involves some possible refactoring. This I would like to keep to separate PRs.

mmzx committed:
Add lts-13.19 with ghc-8.6.4.

mmzx committed:
Increase higher version bound of package: process-1.6.5.

mmzx committed:
Conduit interface changes.

mmzx committed:
use_process_jobs from process-1.5.0.

mmzx committed:
Use minimum wai-3.2.2.

tolysz commented:
If you create one more branch, and one more PR, but directly from `modernize`, and only add `.travis.yaml`, I would have a look. The current changes remove my changes e.g. per bundle ssl cert handling (where a bundle can specify which certificates to use), or use of session  manager... or `process` 1.6... and more...

mmzx commented:
Thanks @tolysz !
I am rather going ground-up: adding more recent ghc versions to the CI and fixing each warning, library incompatibility issue one-by-one.

At the same time I try to make small PRs in the main repository...
So, eventually I will get to the point where all of your changes will be included.

tolysz commented:
I see. I guess that if I will "borrow" your CI and that's it. It should work for me.

mmzx committed:
Use fsnotify-3.0 in each build.

mmzx committed:
Update upper-bound version of package: process.

mmzx committed:
Remove lts-9.21 support from CI.

mmzx committed:
Test CI against several ghc versions.

Syntax correction.

Add ghc-865.

Try with cabal 2.4.

Remove cabal-3.0 and ghc-865 builds.

mmzx commented:
@tolysz , now this one includes
- all your changes from the past (in the form of a squashed merge 
- I think, a quiet good CI script to have extensive test across ghc versions and platforms, also easy to expand.
- the **process** package is not utilized via compiler macros any more (I don't see it a scalable, long-term solution). - and so, **process-3.0.x** can now be used with its latest available API.

mmzx committed:
Enable cabal-install execution in CI.
